### ISSUE TITLE
K3OS Operator

### ISSUE THREAD

**Issue by DblK** [rancher/k3os#157]
In the readme, it mentions an operator for K3OS to be able to update and maybe change K3S version (also talked in the presentation video).

Can you share more information about this operator? Design ideas?
Are you working on it? Do you need help?

I wish to be able to update K3s inside my current K3Os version. For now, since it's a testing env, I can destroy and rebuild everything for update but it's not very nice for production.

**Comment by corvus-ch** [comment]
In the meantime, the following might work to at least change the configuration:

```
mkdir -p /mnt
mount /dev/sda1 /mnt
# Edit /mnt/k3os/system/config.yaml
reboot
```

**Comment by zimme** [comment]
Tracked in #253

**Comment by dweomer** [comment]
I've got a preview of my work-in-progress, addressing upgrades, up on my personal fork: https://github.com/dweomer/k3os/blob/preview/operator/preview.md

**Comment by dweomer** [comment]
For the time being we are treating k3OS as a bundle which means that to get an upgraded version of k3s you'll need to upgrade to a a version of k3os that bundles it. See #284 for more detail.

**Comment by dweomer** [comment]
I should also mention that before #284 (or similar) is merged you can and have been able to upgrade k3s since v0.5.0 (released a week or so before Halloween 2019) via `k3os-upgrade-rootfs` which will also upgrade k3os.

The `k3os-upgrade-[rootfs|kernel]` scripts are being honored as much as is reasonable (so far, no divergence) on my operator branch/pr but I will call out if there end up being any breaking changes.

**Comment by zimme** [comment]
Maybe the `k3os-upgrade-[rootfs|kernel]`  scripts should be deprecated in favor of `k3os ops`?

**Comment by dweomer** [comment]
> Maybe the `k3os-upgrade-[rootfs|kernel]` scripts should be deprecated in favor of `k3os ops`?

Probably. So far the feel of the new CLI is very similar so there wouldn't be much, if anything, lost.

**Comment by bhale** [comment]
> > Maybe the `k3os-upgrade-[rootfs|kernel]` scripts should be deprecated in favor of `k3os ops`?
> 
> Probably. So far the feel of the new CLI is very similar so there wouldn't be much, if anything, lost.

Seems so. +1 to drop the scripts, they were only around for ~2 releases so no one should be too put out.

**Comment by dweomer** [comment]
Reviewing this thread it looks to be primarily concerned with upgrades to a running system. This has been addressed with the most recent release [v0.9.0](https://github.com/rancher/k3os/releases/tag/v0.9.0). Please re-open or create anew to track something that I may have missed.
